src/dbnode/storage/limits/query_limits.go

```diff
@@ -199,7 +237,15 @@ func (q *lookbackLimit) exceeded() error {
 }
 
 func (q *lookbackLimit) checkLimit(recent int64) error {
-	if q.options.Limit > 0 && recent > q.options.Limit {
+	q.lock.RLock()
+	limit := q.options.Limit
```

Comment:
is a lock necessary here if we grab a snapshot and use the snapshot? I imagine there will be a ton of activity on this

Reply:
it's just a read lock, which should be fine. and that's basically what is happening, snapshoting the current value with the lock.